app/models/search.rb

```diff
@@ -0,0 +1,68 @@
+class Search < ActiveRecord::Base
+  require 'date'
+  include ActiveModel::ForbiddenAttributesProtection
+
+  attr_accessible :key_words, :title, :main_type, :note_type, :min_date, :created_by,
+                  :language, :max_date
+
+  ## current advanced search
+  def advanced_search(input, params)
+    all = !params[:notes] && !params[:wikis] && !params[:maps] && !params[:comments]
+    @nodes = []
+    unless input.blank?
+      @nodes += SearchService.new.find_notes(input, 25) if params[:notes] || all
```

Comment:
@jywarren , 
Actually this line of code is not using in latest update. We can remove this piece of code. Now advanced search page completely rely on solr.